packages/cli/src/commands/setup/auth/templates/dbAuth.auth.ts.template

```diff
@@ -20,7 +20,11 @@ import { db } from './db'
  * fields to the `select` object below once you've decided they are safe to be
  * seen if someone were to open the Web Inspector in their browser.
  */
-export const getCurrentUser = async (session: DbAuthSession<number>) => {
+export const getCurrentUser = async (session: Decoded) => {
+  if (!session || typeof session.id !== 'number') {
```

Comment:
Question rather than a comment, what happens when we return null here? 

Isn't the expected behaviour that the findUnique would error out (forcing you to login again)?

Reply:
`findUnique` will return `null` if nothing is found

So, previously, if no session was passed in it would throw. "Uncaught TypeError: Cannot read properties of undefined (reading 'id')"
If an id that doesn't exist was passed in `null` would be returned.

Now I changed it to always return `null`.

Reply:
I could change it to
```
export const getCurrentUser = async (session: Decoded) => {
  if (!session || typeof session.id !== 'number') {
    throw new Error('Invalid session')
  }

  return await db.user.findUnique({
    where: { id: session.id },
    select: { id: true },
  })
}
```

But that's also not how it behaved previously

Reply:
Let me walk through what I'm thinking

1. getCurrentUser is **not** called if auth headers do not exist i.e. for public requests

So gCU is only called when it is an authenticated request (https://github.com/dac09/redwood/blob/c6b6732492999833b1a1b5dad42908682f23c54e/packages/graphql-server/src/plugins/useRedwoodAuthContext.ts#L35-L38) 

So if its an authenticated request... if session or sessionId don't exist, it should be an error no? 

I also wouldn't do the check for number here, because the id might be a string, based on the User db model - what's the reason for adding this extra check anyway? TS being TS?

Reply:
Also... It wouldn't reach the findUnique code, it would error out trying to access undefined.id - and also error out if you supplied a string instead of number.

Reply:
@cannikin HELP

Reply:
We "checked" that it was a number before too, by the type `DbAuthSession<number>`
We generate the code expecting the id will be a number. If the user changes that to something else I don't think it's too much to expect they'll have to also update some generated code. And TS will help them find it too, because they'll get type errors here.

Without the `typeof` check the type of `session.id` is `unknown` and Prisma's TS types don't like that.

Reply:
Oh but you forget people also write JavaScript. Types get stripped out, code like this doesn't.

Reply:
> Oh but you forget people also write JavaScript. Types get stripped out, code like this doesn't.

No, I'm not forgetting 😃  I'd say that's an even bigger reason to have this code than to just rely on types 🙂 

I had to try and see what would happen if you had an `id` of the wrong type.

![image](https://user-images.githubusercontent.com/30793/187993571-debc0258-dcda-443e-b16e-46c6f2580a87.png)

So let's try to keep the behavior of throwing an error if the type is wrong.

In `useRedwoodAuthContext` I'm seeing this

![image](https://user-images.githubusercontent.com/30793/187994392-d076e8bc-ce55-4f1e-b3e2-52add8ca3c56.png)

Why would the execution stop if `authContext[0]` is `null`?

Reply:
Also, returning `null` if we don't have a session is what a lot of the other auth providers do. Here's the base implementation

![image](https://user-images.githubusercontent.com/30793/187995281-b942e626-88e5-4792-a24d-10ccef890a53.png)

Reply:
I’d think this should return `null` if the user isn’t logged in. Not being logged in is a totally normal flow, not something that should error. Having code like:

```js
if (currentUser) {
  // …
}
```

Should be totally valid. I don’t think someone should HAVE to check `isAuthenticated`, just find out from `currentUser`.

If it wasn’t already returning `null` that was my bad!

Reply:
I'm ok with it if you are Rob, but just remember that getCurrentUser isn't called for not-logged-in-flows. It's only called when the headers for auth are present i.e. I-think-Im-logged-in-flows.

**Before your changes:**
> Why would the execution stop if authContext[0] is null?
If its null, it will error out accessing `null.id`.  Because well.. you can't access null['id'] or undefined['id'].

> If its a string instead of number
You see the prisma error you showed in the screenshot. 


With your new code, neither of these errros will be thrown. This is particularly important if you start with userId as a number, then decide I want to make it a string.

All of this is moot though if throwing an error in currentUser is exactly the same as returning null from currentUser. 

I'll defer to your decision on this, no point going round in circles. If it breaks, it breaks.

Reply:
Agreed. Feels like we're going round in circles here on what the exact implementation should be.
What if we instead take a step back and think about what we're worried about/what we want the experience to be

 1. We want the code to work (duh!)
    * No matter if you're logged in or not, you should be able to call `getCurrentUser()`. 
    * You should always be able to check `isAuthenticated` and `currentUser` 
 2. Helpful error message if the user starts out with numeric user ids, but later switches to strings, and forgets to update the code.
 
 I'll run some experiments and then decide on what I think is the best path forward

Reply:
> I'm going to go with throwing an error

Actually pushing commits help... https://github.com/redwoodjs/redwood/pull/6340